src/BenchmarksApps/SslStream/Client/Program.cs

```diff
@@ -0,0 +1,249 @@
+
+        static async Task DoRun(ClientOptions options)
+        {
+            while (s_isRunning)
+            {
+                using var sock = new Socket(SocketType.Stream, ProtocolType.Tcp);
+                await sock.ConnectAsync(options.Hostname, options.Port).ConfigureAwait(false);
+                await using var stream = await EstablishSslStreamAsync(sock, options).ConfigureAwait(false);
+                s_counters.TotalHandshakes++;
+            }
+        }
+
+        s_isRunning = true;
+        var task = Task.Run(() => DoRun(options));
+        await Task.Delay(options.Warmup).ConfigureAwait(false);
+        s_isRunning = false;
+        Log("Completing warmup...");
+        await task.ConfigureAwait(false);
+
+        s_counters.Reset();
+
+        s_isRunning = true;
+        task = Task.Run(() => DoRun(options));
+        Stopwatch sw = Stopwatch.StartNew();
+        await Task.Delay(options.Duration).ConfigureAwait(false);
+        s_isRunning = false;
+        Log("Completing scenario...");
+        await task.ConfigureAwait(false);
+        sw.Stop();
+
+        LogMetric("sslstream/handshake/mean", sw.Elapsed.TotalMilliseconds / s_counters.TotalHandshakes);
+    }
+
+    static async Task RunReadWriteScenario(ClientOptions options)
+    {
+
+        static async Task WritingTask(SslStream stream, int bufferSize, CancellationToken cancellationToken = default)
+        {
+            if (bufferSize == 0)
+            {
+                return;
+            }
+
+            BenchmarksEventSource.Register("sslstream/write/mean", Operations.Avg, Operations.Avg, "Mean bytes written per second.", "Bytes per second - mean", "n2");
+
+            var sendBuffer = new byte[bufferSize];
+            Stopwatch sw = Stopwatch.StartNew();
+            bool isWarmup = true;
+
+            while (s_isRunning)
+            {
+                if (isWarmup && !s_isWarmup)
+                {
+                    isWarmup = false;
+                    sw.Restart();
+                    s_counters.BytesWritten = 0;
+                }
+
+                await stream.WriteAsync(sendBuffer, cancellationToken).ConfigureAwait(false);
+                s_counters.BytesWritten += bufferSize;
+            }
+
+            sw.Stop();
+            LogMetric("sslstream/write/mean", s_counters.BytesWritten / sw.Elapsed.TotalSeconds);
+        }
+
+        static async Task ReadingTask(SslStream stream, int bufferSize, CancellationToken cancellationToken = default)
+        {
+            if (bufferSize == 0)
+            {
+                return;
+            }
+
+            BenchmarksEventSource.Register("sslstream/read/mean", Operations.Avg, Operations.Avg, "Mean bytes read per second.", "Bytes per second - mean", "n2");
+
+            var recvBuffer = new byte[bufferSize];
+            Stopwatch sw = Stopwatch.StartNew();
+            bool isWarmup = true;
+
+            while (s_isRunning)
```

Comment:
what is typical duration? We should make sure the read/write phase is much longer than handshake

Reply:
or would that happen in warmup phase?

Reply:
I copied default values from HttpClient benchmarks, both warmup and duration are 15s by default.

Reply:
handshake is not accounted for in either of those in the readwrite scenario